hfcufb.js

```diff
@@ -1075,10 +1075,22 @@ function validSource(entry) {
 function setupOptions(){
 
     factionSelection = document.getElementById("factionSelector");
+
     factions.forEach(faction =>{
         var option = new Option(displayText[faction], faction);
-        factionSelection.add(option);
+        var alreadyExists = false;
```

Comment:
With the introduction of factions beyond Kushan and Taiidan, this might not be sufficient. I want to keep the ability to toggle content packs on and off. With this block, if a faction is added, it won't be removed. This isn't a big deal, though, it can be added in later.

Reply:
yeah, that makes sense, I had similar thoughts which is why I created [issue 10](https://github.com/achapin/homeworldfleetcommand-forcebuilder/issues/10).

Also, supporting an arbitrary number of factions would require some changes as the data file structure and parsing as currently they explictly list all applicable factions for each unit in an array instead of simply listing, e.g., "all" if that's the case (e.g., neutral cards). So any addition of a faction requires rework of the data files, which is suboptimal.